collect_app/src/main/java/org/odk/collect/android/dao/FormsDao.java

```diff
@@ -129,6 +129,26 @@ public void deleteFormsFromIDs(String[] idsToDelete) {
         Collect.getInstance().getContentResolver().delete(FormsProviderAPI.FormsColumns.CONTENT_URI, selection, idsToDelete);
     }
 
+    public void deleteFormsFromMd5Hash(String[] toDelete) {
```

Comment:
Using varargs will simplify calling the method. Use new style for. Since you have a loop, you need to close the cursor each time.

![delete forms changes](https://user-images.githubusercontent.com/55099/35612221-d218d150-061c-11e8-9ead-5e53e8b22e12.png)